src/components/Drawer/Drawer.tsx

```diff
@@ -0,0 +1,40 @@
+import { Box, IconButton, Button, Divider, SwipeableDrawer } from '@mui/material';
+import ChevronRightIcon from '@mui/icons-material/ChevronRight';
+import DrawerHeader from './DrawerHeader';
+import Logo from '../Logo/Logo';
+import List from '../List/List';
+import { toggleDrawer } from './utils';
+import { DrawerT } from '../../types/types';
+
+const Drawer = ({ open, setOpen, navigate, siteTreeOptions }: DrawerT) => {
```

Comment:
Is it necessary to pass `navigate` as a prop?

Couldn't the components needing to use it get it inside them with the `useNavigate`?